### PR TITLE
Strip quotes from debug configurations and fix an issue with quotes in paths

### PR DESCRIPTION
**Pull request by bobbrow**
This should address a few CodeQL issues flagged in the repo. There was some code added to escape any double quotes that might be added from the generated debug configurations during the localization process. Since we don't really want those double quotes anyway, it should be acceptable to simply remove them.

While looking at some of the other issues flagged, I noticed that a helper function ends up thinking a quoted path is a relative path. Fixing that doesn't fix any other underlying assumptions we make about the include paths set by the user, but it does allow them to be squiggled when the true path is invalid.

### CODE REVIEW

File: Extension/src/LanguageServer/configurations.ts [hunk 1]
```diff
@@ -1998,7 +2006,6 @@ export class CppProperties {
             }
 
             // Escape the path string for literal use in a regular expression
-            // Need to escape any quotes to match the original text
             let escapedPath: string = curPath.replace(/"/g, '\\"');
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/configurations.ts]: What about this case?

**Review by sean-mcmanus — DISMISSED** [review]
Can you get @WardenGnaw to approve too?

File: Extension/src/LanguageServer/configurations.ts (hunk 1 above)
Comment (bobbrow) [reply Extension/src/LanguageServer/configurations.ts]: That's the one I dismissed

Comment (bobbrow) [reply Extension/src/LanguageServer/configurations.ts]: oh, but I didn't mean to remove the comment. That was left over from my poking around at it.

**Review by sean-mcmanus — APPROVED** [review]

**Comment by bobbrow** [comment]
> Can you get @WardenGnaw to approve too?

I'll ask him, but we don't need to block on it since they're not active here.

**Review by sean-mcmanus — APPROVED** [review]

**Comment by WardenGnaw** [comment]
It looks like these changes came from this commit: https://github.com/microsoft/vscode-cpptools/commit/b82a86e5e4b183b13c1db2e4e4149232927f7004

Which was to fix https://github.com/microsoft/vscode-cpptools/issues/4526

I would do a quick validation in a German VS Code to ensure the debug configuration is still being generated properly.

**Review by WardenGnaw — APPROVED** [review]

**Comment by bobbrow** [comment]
> It looks like these changes came from this commit: [b82a86e](https://github.com/microsoft/vscode-cpptools/commit/b82a86e5e4b183b13c1db2e4e4149232927f7004)
> 
> Which was to fix #4526
> 
> I would do a quick validation in a German VS Code to ensure the debug configuration is still being generated properly.

I tested it by modifying the strings directly in TypeScript. It should be the same case as having the localizers put a quote in the translation.

**Pull request merged**